### ISSUE TITLE
extractAllToAsync error

### ISSUE THREAD

**Issue by lucy1564351** [cthackers/adm-zip#484]
![微信图片_20240424111657](https://github.com/cthackers/adm-zip/assets/134355235/5f6435f5-cfaa-4124-a436-1b30523b0b30)
Never execute   “callback()”

**Comment by 100tomer** [comment]
+1

**Comment by 100tomer** [comment]
For me it worked just switching those lines.. but I don't want to fork this great npm package. Do you know maybe when this changed happened? because I did work for me like a month ago

**Comment by troylu8** [comment]
Here is a simple self contained example, using adm-zip@0.5.12

When `overwrite` is true, the callback is never called. It worked for me with `overwrite` off
```js
const Zip = require('adm-zip');

const zip = new Zip("./myzip.zip");

zip.extractAllToAsync("./dest", 
    true, // when overwrite is true, callback is never called. 
    false, 
    (err) => console.log("callback called")
);
```

**Comment by Linya-IronMan** [comment]
+1

**Comment by Jellfedora** [comment]
+1

**Comment by Jellfedora** [comment]
> Voici un exemple simple et autonome, utilisant adm-zip@0.5.12
> 
> Lorsque `overwrite`c'est vrai, le rappel n'est jamais appelé. Cela a fonctionné pour moi `overwrite`sans
> 
> ```js
> const Zip = require('adm-zip');
> 
> const zip = new Zip("./myzip.zip");
> 
> zip.extractAllToAsync("./dest", 
>     true, // when overwrite is true, callback is never called. 
>     false, 
>     (err) => console.log("callback called")
> );
> ```

True or false dont work for me after upgrade 0.5.10 to 0.5.12

**Comment by Jellfedora** [comment]
> Here is a simple self contained example, using adm-zip@0.5.12
> 
> When `overwrite` is true, the callback is never called. It worked for me with `overwrite` off
> 
> ```js
> const Zip = require('adm-zip');
> 
> const zip = new Zip("./myzip.zip");
> 
> zip.extractAllToAsync("./dest", 
>     true, // when overwrite is true, callback is never called. 
>     false, 
>     (err) => console.log("callback called")
> );
> ```

True or false dont work for me after upgrade 0.5.10 to 0.5.12

**Comment by ZoltanT-RD** [comment]
+1

**Comment by ahaubenstock** [comment]
+1

**Comment by 5saviahv** [comment]
fixed in repo

**Comment by Arkellys** [comment]
Fix released on [v0.5.13](https://github.com/cthackers/adm-zip/releases/tag/v0.5.13)
